src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -935,13 +937,13 @@ private void ValidateState(CompilationTrackerState? state)
                 {
                     ValidateCompilationTreesMatchesProjectState(finalState.FinalCompilationWithGeneratedDocuments, ProjectState, finalState.GeneratorInfo);
                 }
-                else if (state is InProgressState inProgressState)
+                else if (state is InProgressState { PendingTranslationSteps.Count: > 0 } inProgressState)
```

Comment:
You can probably tweak this a bit and remove this condition, and then update the uses of `inProgressState.PendingTranslationSteps[0].oldState` to be `inProgressState.PendingTranslationSteps[0].oldState ?? this.ProjectState`: basically, once we've completed all the translation steps we expect the syntax trees in the compilation to match all the various documents we have.

Reply:
k

Reply:
done.